apps/addons/models.py

```diff
@@ -497,7 +494,14 @@ def delete(self, msg='', reason=''):
     @classmethod
     def initialize_addon_from_upload(cls, data, is_listed=True):
         fields = cls._meta.get_all_field_names()
+        # Reclaim GUID from deleted add-on.
+        try:
+            old_guid_addon = Addon.unfiltered.get(guid=data['guid'])
+            old_guid_addon.update(guid=None)
```

Comment:
We don't know what the pk is until Addon is created and .save() is called to write to the database.  If someone deletes the add-on several times each time the pk will be different so won't have the same guid-reused-by-pk-1234 string.